### PR TITLE
Replace jwkest with authlib

### PR DESCRIPTION
**Pull request by kleesc**
**Issue:** https://issues.redhat.com/browse/PROJQUAY-224

Follow up to #679 

**Changelog:** 
- Replaces jwkest (deprecated, and depends on `PyCrypto`) with authlib (supports multiple backends, including `cryptography`) 

**Docs:** 

**Testing:** 

**Details:**

### CODE REVIEW

**Comment by lepture** [comment]
@kleesc Authlib has an alpha 1.0.0 release, which support generate key natively. https://docs.authlib.org/en/latest/specs/rfc7517.html#authlib.jose.JsonWebKey.generate_key

File: boot.py [hunk 1]
```diff
@@ -94,9 +94,15 @@ def setup_jwt_proxy():
...
             f.write(quay_key_id)
 
-        with open(app.config["INSTANCE_SERVICE_KEY_LOCATION"], mode="w") as f:
+        with open(app.config["INSTANCE_SERVICE_KEY_LOCATION"], mode="wb") as f:
             f.truncate(0)
-            f.write(quay_key.exportKey().decode("utf-8"))
+            f.write(
+                quay_key.private_bytes(
+                    encoding=serialization.Encoding.PEM,
+                    format=serialization.PrivateFormat.TraditionalOpenSSL,
```
Comment (thomasmckay) [inline boot.py:102]: missing `from cryptography.hazmat.primitives import serialization`

**Review by thomasmckay — CHANGES_REQUESTED** [review]

File: requirements-osbs.txt [hunk 2]
```diff
@@ -87,7 +87,7 @@ mixpanel==4.5.0
 mock==3.0.5
 mockldap @ git+https://github.com/quay/mockldap.git@4265554a3d89fe39bf05b18e91607bec3fcf215a
 monotonic==1.5
-moto==1.3.14
+moto==2.0.1
```
Comment (thomasmckay) [inline requirements-osbs.txt:90]: Remove this as it tries to bring in other deps and we don't need it at runtime. I've created PROJQUAY-1720 to track revisiting all the packages.

**Review by thomasmckay — CHANGES_REQUESTED** [review]

**Review by thomasmckay — APPROVED** [review]
LGTM

**Pull request merged**